res/controllers/midi-components-0.0.js

```diff
@@ -664,22 +664,19 @@
     };
 
     var Deck = function(deckNumbers) {
-        if (deckNumbers !== undefined) {
-            if (Array.isArray(deckNumbers)) {
-                // These must be unique to each instance,
-                // so they cannot be in the prototype.
-                this.currentDeck = "[Channel" + deckNumbers[0] + "]";
-                this.deckNumbers = deckNumbers;
-            } else if (typeof deckNumbers === "number" &&
-                      Math.floor(deckNumbers) === deckNumbers &&
-                      isFinite(deckNumbers)) {
-                this.currentDeck = "[Channel" + deckNumbers + "]";
-                this.deckNumbers = [deckNumbers];
-            }
+        if (deckNumbers !== undefined && Array.isArray(deckNumbers)) {
+            // These must be unique to each instance,
+            // so they cannot be in the prototype.
+            this.deckNumbers = deckNumbers;
+        } else if (deckNumbers !== undefined && typeof deckNumbers === "number" &&
```

Comment:
Can we skip checking `deckNumbers !== undefined` twice?

Reply:
You have an idea how? I introduced that check in 6f0c690 because according to Be the QtScriptEngine behaves weird if it's undefined: https://github.com/mixxxdj/mixxx/pull/2948/files/21cafe18eba551ddb970299fef40458b6e2adb40#diff-9bedde251ff7d14d89b40d05168275e4

Reply:
I was just thinking about extracting that into its own check like there was before.
```javascript
if (deckNumbers !== undefined) {
  if (Array.isArray(deckNumbers)) { ... }
  else if (typeof deckNumbers === "number" && ...) {...}
  }
}
```

Reply:
And in case deckNumbers is not undefined and neither an array nor a number?

Reply:
Ah, right, that would result in more duplicated code. You are right, the code is fine as-is.